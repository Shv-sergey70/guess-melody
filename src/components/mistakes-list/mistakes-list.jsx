import React from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {getMistakes} from "../../reducer/game/selectors";

const MistakesList = ({mistakes}) => {
  const content = new Array(mistakes)
    .fill(``)
    .map((item, i) => <div key={i} className="wrong"/>);

  return (
    <div className="game__mistakes">
      {content}
    </div>
  );
};

MistakesList.propTypes = {
  mistakes: PropTypes.number.isRequired
};

const mapStateToProps = (state) => ({
  mistakes: getMistakes(state)
});

export {MistakesList};
export default connect(mapStateToProps)(MistakesList);
