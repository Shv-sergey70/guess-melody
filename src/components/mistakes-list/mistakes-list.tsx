import * as React from 'react';
import {connect} from "react-redux";
import {getMistakes} from "../../reducer/game/selectors";

type Props = {
  mistakes: number
};

const MistakesList: React.FunctionComponent<Props> = ({mistakes}) => {
  const content = new Array(mistakes)
    .fill(``)
    .map((item, i) => <div key={i} className="wrong"/>);

  return (
    <div className="game__mistakes">
      {content}
    </div>
  );
};

const mapStateToProps = (state) => ({
  mistakes: getMistakes(state)
});

export {MistakesList};
export default connect(mapStateToProps)(MistakesList);
