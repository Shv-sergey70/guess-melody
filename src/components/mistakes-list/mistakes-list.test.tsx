import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {MistakesList} from "./mistakes-list";

configure({adapter: new Adapter()});

test(`MistakesList correctly renders`, () => {
  const tree = shallow(<MistakesList mistakes={2} />);

  expect(tree).toMatchSnapshot();
});
