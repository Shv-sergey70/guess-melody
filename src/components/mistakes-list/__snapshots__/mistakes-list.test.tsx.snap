// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MistakesList correctly renders 1`] = `ShallowWrapper {}`;
